example.py

```diff
@@ -43,7 +43,8 @@ def simulate(car, time_step=0.0002, interval=1):
 
 
 if __name__ == "__main__":
-    car = Car()
+    road = Road(length= 6, mode="square")
+    car = Car(road_func=road)
```

Comment:
Explicitly creating the `Road` instance and passing it to `Car` like you've done here is probably a good idea (to demonstrate how to pass a road object to `Car`). Was that part of your intent? Currently, `Car` instantiates a `Road` object with certain default parameters if `road_func` was not passed to `Car.__init__()` (https://github.com/nrsyed/half-car/blob/0664f0a641600c63f2d65eb2925829abeb81d3a8/halfcar/car.py#L274-L284). The road `length` and `x_min` are based on the vehicle's properties, specifically `l_r` and `l_f`. I prefer this to hard-coding values (e.g., `length=6).

Let me know if this changes your thoughts.

Reply:
Currently creating a Road required x_min

```
road = Road(length= 6, x_min=-3, mode="bump", frequency=1.4, amplitude= 0.1)
```

omitting this gives

> ValueError: A value in x_new is below the interpolation range.